CONTRIBUTING.md

```diff
@@ -45,9 +45,90 @@ include your changes.
   the name of the thing you're changing in at the beginning of the message,
   followed by a colon: the plugin name for plugins, "docs" for documentation
   files, "coretasks" for `coretasks.py`, "db" for the database feature, etc.
-* Python files should always have  `from __future__ import generator_stop`
+* Python files should always have `from __future__ import annotations`
   as the first line after the module docstring.
 
+Running from source
+-------------------
+
+Sopel is your typical Python project: you can install it from source, edit
+the code using standard tooling, and take advantage of your favorite code
+editor.
+
+Assuming you are using standard tools:
+
+* Create a virtualenv.
+  * You can use `virtualenv` (or the `venv` built-in) directly,
+    `virtualenvwrapper`, or any tool that allows you to create and manage your
+    virtualenvs.
+  * Your project manager may create the virtualenv for you, be sure to check
+    the documentation of the tools you are using.
+* Fork Sopel's repository, and clone your fork locally.
+* Activate your virtualenv, and `cd` into your clone's folder.
+* Ensure you have the latest version of `pip`, and install `wheel`.
```

Comment:
Is explicitly installing `wheel` still necessary? It and `setuptools` are declared as `build-system` dependencies in `pyproject.toml`.

Reply:
`wheel` is never necessary, it is only convenient: if installed, it means that doing `pip install -r dev-requirements.txt` for example will be able to generate wheels for the downloaded packages that don't provide them, even tho `pip install -e .` wasn't done yet.

So yeah, I consider that good advice anyway! It's like upgrading `pip` and installing `pytz`: these should be almost automatic when starting a new virtualenv.